lib/newman_conway.js

```diff
@@ -1,10 +1,26 @@
 
-// Time Complexity:
-// Space Complexity:
+// Time Complexity: O(n)
+// Space Complexity: O(n)
 
 function newmanConway(num) {
-  throw new Error("Function not implemented yet...")
-}
+  if (num === 0) throw new Error();
+
+  // let i;
+  let array = [];
+  array[0] = 0;
+  array[1] = 1;
+  array[2] = 1;
+
+  if (num === 1) return '1';
+
+  for (let i = 3; i <= num; i++) {
+    array[i] = array[array[i - 1]] + array[i - array[i - 1]];
```

Comment:
In general, be careful assigning to arbitrary indices in an array that haven't been initializes. Yes, JS will grow the array to fill any missing intervening indices, but it can have surprising effects. Here, since we're adding each next index with no gaps, everything works as expected, but consider using `push` to make it more obvious that we are growing the array here.